src/output-json-alert.c

```diff
@@ -310,6 +314,16 @@ static void AlertAddAppLayer(const Packet *p, JsonBuilder *jb,
                 if (!al->LogTx(tx, jb)) {
                     jb_restore_mark(jb, &mark);
                 }
+                if (proto == ALPROTO_WEBSOCKET) {
+                    if (option_flags &
+                            (LOG_JSON_WEBSOCKET_PAYLOAD | LOG_JSON_WEBSOCKET_PAYLOAD_BASE64)) {
+                        bool pp = (option_flags & LOG_JSON_WEBSOCKET_PAYLOAD) != 0;
+                        bool pb64 = (option_flags & LOG_JSON_WEBSOCKET_PAYLOAD_BASE64) != 0;
+                        if (!SCWebSocketLogDetails(tx, jb, pp, pb64)) {
+                            jb_restore_mark(jb, &mark);
+                        }
+                    }
+                }
```

Comment:
I guess here we've already opened, logged and closed the app-layer TX.  It does feel like a hack.  I think I would rather see:
- the opening and closing of the object here, then LogTx just fills in the open object but doesn't make the choice about closing it, somewhat undoes some previous work you did though.
- or passing in extra flags to LogTx

Reply:
First, I had the flags to the logging function, but it did not feel right.

I will switch back to that in the next version so that we can talk about it